platforms/documentation/docs/src/docs/userguide/dep-man/declaring-depedencies/declaring_configurations.adoc

```diff
@@ -0,0 +1,189 @@
+
+====
+include::sample[dir="snippets/dependencyManagement/attributeMatching/kotlin",files="build.gradle.kts[tags=java-configuration-example]"]
+include::sample[dir="snippets/dependencyManagement/attributeMatching/groovy",files="build.gradle[tags=java-configuration-example]"]
+====
+
+The following configurations are created:
+
+- `implementation`: Used for declaring project dependencies but neither consumed nor resolved.
+- `compileClasspath` + `runtimeClasspath`: Resolvable configurations that collect compile-time and runtime dependencies from `implementation`.
+- `exposedApi` + `exposedRuntime`: Consumable configurations that expose artifacts (API and runtime) to other projects, but aren’t meant for internal resolution.
+
+This setup mimics the behavior of the `implementation`, `compileClasspath`, `runtimeClasspath`, `apiElements`, and `runtimeElements` configurations in the <<java_plugin.adoc#java_plugin,Java plugin>>.
+
+[[sec:defining-custom-configurations]]
+== Creating custom configurations
+
+You can define custom configurations to declare separate scopes of dependencies for specific purposes.
+
+Suppose you want to generate Javadocs with AsciiDoc formatting embedded within your Java source code comments.
+By setting up the `asciidoclet` configuration, you enable Gradle to use Asciidoclet, allowing your Javadoc task to produce HTML documentation with enhanced formatting options:
+
+====
+include::sample[dir="snippets/java/basic/kotlin",files="build.gradle.kts[tags=create-custom-doclet]"]
+include::sample[dir="snippets/java/basic/groovy",files="build.gradle[tags=create-custom-doclet]"]
+====
+
+You can manage custom configurations using the `configurations` block.
+Configurations can have names and can extend each other.
+For more details, refer to the link:{groovyDslPath}/org.gradle.api.artifacts.ConfigurationContainer.html[`ConfigurationContainer`] API.
+
+NOTE: Configurations are intended to be used for a single role: declaring dependencies, performing resolution, or defining consumable variants.
+
+There are three main use cases for creating custom configurations:
+
+1. **API/Implementation Separation:** Create custom configurations to separate API dependencies (exposed to consumers) from implementation dependencies (used internally during compilation or runtime).
+- You might create an `api` configuration for libraries that consumers will depend on, and an `implementation` configuration for libraries that are only needed internally. The `api` configuration is typically consumed by downstream projects, while `implementation` dependencies are hidden from consumers but used internally.
+- This separation ensures that your project maintains clean boundaries between its API and internal dependencies.
+
+2. **Resolvable Configuration from Dependency Configuration**: Create a custom resolvable configuration to resolve specific sets of dependencies, like classpaths, at various build stages.
```

Comment:
🤔 I don't understand the "from Dependendency Configuration" title here.  What is this trying to say?  Everything after that makes sense.